framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -405,19 +413,27 @@ player_result_t MediaPlayerImpl::setVolume(int vol)
 	return ret;
 }
 
-void MediaPlayerImpl::setVolumePlayer(int vol, player_result_t &ret)
+void MediaPlayerImpl::setVolumePlayer(uint8_t vol, player_result_t &ret)
 {
-	int vol_max = -1;
-	medvdbg("MediaPlayer Worker : setVolume\n");
+	uint8_t vol_max;
+	medvdbg("MediaPlayer Worker : setVolume %d\n", vol);
+
+	audio_manager_result_t result = get_max_audio_volume(&vol_max);
+	if (result != AUDIO_MANAGER_SUCCESS) {
+		meddbg("To set the output volume is not supported in the current audio device\n");
+		ret = PLAYER_DEVICE_NOT_SUPPORT;
+		goto errout;
+	}
 
-	vol_max = get_max_audio_volume();
-	if (vol < 0 || vol > vol_max) {
+	if (vol > vol_max) {
```

Comment:
I think like a mobile device, if volume up requested if current state is max volume, then just return OK will be more good choice, how do you think?